File: backend/api/routers/v1/integration.py
from fastapi import APIRouter

from api.core.config import settings
from api.core.impl import auth_backend
from api.core.model_catalog import DEFAULT_MODEL_KEY, frontend_model_options
from api.schemas.integration import FrontendConfig


router = APIRouter(prefix='/integration', tags=['integration'])


@router.get('/frontend')
async def frontend_config() -> FrontendConfig:
    return FrontendConfig(
        auth_enabled=bool(auth_backend),
        key_predefined=settings.BACKEND_STATIC_OAI_KEY is not None or settings.BACKEND_USE_PROXY_STATIC_KEY,
        default_model=DEFAULT_MODEL_KEY,
        models=frontend_model_options(),
    )

File: frontend/src/lib/integration.ts
import { API_BASE } from "@/lib/api"

export interface FrontendModelOption {
  key: string
  label: string
}

export const DEFAULT_MODEL_OPTIONS: FrontendModelOption[] = [
  { key: "codex-gpt-5.5", label: "codex-gpt-5.5" },
  { key: "codex-gpt-5.4", label: "codex-gpt-5.4" },
  { key: "codex-gpt-5.4-mini", label: "codex-gpt-5.4-mini" },
  { key: "codex-gpt-5.2", label: "codex-gpt-5.2 (legacy)" },
  {
    key: "codex-gpt-5.1-codex-max",
    label: "codex-gpt-5.1-codex-max (legacy)",
  },
]

export const DEFAULT_MODEL_KEY =
  DEFAULT_MODEL_OPTIONS[0]?.key ?? "codex-gpt-5.5"

export interface FrontendConfig {
  auth_enabled: boolean
  key_predefined: boolean
  default_model: string
  models: FrontendModelOption[]
}

export async function fetchFrontendConfig(
  signal?: AbortSignal,
): Promise<FrontendConfig> {
  const response = await fetch(`${API_BASE}/v1/integration/frontend`, {
    signal,
    cache: "no-store",
    credentials: "omit",
  })

  if (!response.ok) {
    throw new Error(`Failed to load frontend config (${response.status})`)
  }

  return response.json()
}
